libs/common-system/src/main/java/org/codice/ddf/configuration/SystemBaseUrl.java

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.configuration;
+
+/**
+ * An immutable utility class for getting system url information. Information is pulled from
+ * the system properties.
+ */
+public final class SystemBaseUrl {
+
+    public static final String HTTP_PORT = "org.codice.ddf.system.httpPort";
+
+    public static final String HTTPS_PORT = "org.codice.ddf.system.httpsPort";
+
+    public static final String HOST = "org.codice.ddf.system.hostname";
+
+    public static final String PROTOCOL = "org.codice.ddf.system.protocol";
+
+    public static final String EXT_HTTP_PORT = "org.codice.ddf.system.externalHttpPort";
+
+    public static final String EXT_HTTPS_PORT = "org.codice.ddf.system.externalHttpsPort";
+
+    public static final String EXT_HOST = "org.codice.ddf.system.externalHostname";
+
+    public static final String ROOT_CONTEXT = "org.codice.ddf.system.rootContext";
+
+    public static final String DEFAULT_HTTP_PORT = "8181";
+
+    public static final String DEFAULT_HTTPS_PORT = "8993";
+
+    public static final String DEFAULT_HOST = "localhost";
+
+    public static final String DEFAULT_PROTOCOL = "https://";
+
+    public static final String DEFAULT_EXT_HTTP_PORT = DEFAULT_HTTP_PORT;
+
+    public static final String DEFAULT_EXT_HTTPS_PORT = DEFAULT_HTTPS_PORT;
+
+    public static final String DEFAULT_EXT_HOST = DEFAULT_HOST;
+
+    public static final String DEFAULT_ROOT_CONTEXT = "/services";
```

Comment:
Removed the default property for root context now that we are using the correct property.